src/components/Personalization/handlers/createHtmlContentHandler.js

```diff
@@ -0,0 +1,44 @@
+/*
+Copyright 2023 Adobe. All rights reserved.
+This file is licensed to you under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License. You may obtain a copy
+of the License at http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software distributed under
+the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+OF ANY KIND, either express or implied. See the License for the specific language
+governing permissions and limitations under the License.
+*/
+import { HTML_CONTENT_ITEM } from "../constants/schema";
+import { VIEW_SCOPE_TYPE } from "../constants/scopeType";
+import isPageWideScope from "../utils/isPageWideScope";
+
+export default ({ next, modules, preprocess }) => proposition => {
+  const {
+    scope,
+    scopeDetails: { characteristics: { scopeType } = {} } = {},
+    items = []
+  } = proposition.getHandle();
+
+  items.forEach((item, index) => {
+    const { schema, data } = item;
+    const { type, selector } = data || {};
+    if (schema === HTML_CONTENT_ITEM && type && selector && modules[type]) {
+      proposition.includeInDisplayNotification();
+      const preprocessedData = preprocess(data);
+      proposition.addRenderer(index, () => {
+        return modules[type](preprocessedData);
+      });
+    }
+  });
+
+  // only continue processing if it is a view scope proposition
+  // or if it is a page wide proposition.
+  if (
+    scopeType === VIEW_SCOPE_TYPE ||
+    isPageWideScope(scope) ||
+    proposition.isApplyPropositions()
+  ) {
```

Comment:
which test case?

Reply:
test/unit/specs/components/Personalization/topLevel/mixedPropositions.spec.js
Specifically, when this check is removed, this proposition is rendered when returned from the server whereas before this refactor, it was not rendered:
```
{
          id: "AT:eyJhY3Rpdml0eUlkIjoiNDQyMzU4IiwiZXhwZXJpZW5jZUlkIjoiIn0=",
          scope: "home",
          items: [
            {
              id: "442358",
              schema: "https://ns.adobe.com/personalization/dom-action",
              data: {
                type: "click",
                format: "application/vnd.adobe.target.dom-action",
                selector: "#root"
              }
            }
          ]
        }
```

Reply:
This looks like a click metric proposition for an SPA view. It should be stored in SPA cache and executed when view change occurs.

Reply:
I figured out the issue here. This proposition shouldn't be rendered because it is not a page wide scope, and it is not a view scopeType. With the changes in #1036, The propositions are separated into three different kinds when fetched from experience edge: "page" - page wide scope or page wide surface, "view" - SPA view scopes, and "proposition" everything else (basically traditional mboxes). Only "page" and "view" types are rendered now.